ESP8266Interface.cpp

```diff
@@ -342,12 +344,23 @@ int ESP8266Interface::socket_close(void *handle)
 
     socket->connected = false;
     _ids[socket->id] = false;
+    _local_ports[socket->id] = 0;
     delete socket;
     return err;
 }
 
 int ESP8266Interface::socket_bind(void *handle, const SocketAddress &address)
 {
+    struct esp8266_socket *socket = (struct esp8266_socket *)handle;
+
+    if (socket->proto == NSAPI_UDP) {
```

Comment:
Should probably fault (whatever the equivalent of EINVAL is?) if the socket has already been used so the module has already chosen a port. (ie `connected` flag set?)

Reply:
ok, so need to add that one